### PR TITLE
[GPII-2951] gpii.net dns zone migration

### PR DESCRIPTION
**Pull request by amatas**
This PR migrates the dns zone from AWS to GCP.

Once this PR is merged and the zone is recreated in Google DNS the NS registries must be modified manually in the register.

### CODE REVIEW

**Comment by natarajaya** [comment]
LGTM

> And as this is a pretty sensitive thing, should we perhaps add some verification steps to the deployment plan

+1, also – will this require downtime for DNS propagation?

**Comment by amatas** [comment]
>  I'm wondering how did you determine which ones are needed and which ones are not?

Ask Gregg.

> And as this is a pretty sensitive thing, should we perhaps add some verification steps to the deployment plan, before proceeding with the NS switch at the domain registrar?
> (for example manually querying several of the most important records like GPII prod against the new GCP NS servers)

>  will this require downtime for DNS propagation?

We can apply this code without any problem because the  zone `gcp.gpii.net` remains untouched, so the DNS from AWS will work.

At that moment we can check with `dig` all the DNS values from the root `gpii.net` to `flowmanager.prd.gcp.gpii.net`. If everything works fine we can switch the NS at the register.

I think that there is no downtime, because if the zones are properly set in GCP and AWS, once the root DNS switch the NS registries the clients will pick up the new registries at GCP so, no downtime is expected.

File: common/modules/gcp-dns/main.tf [hunk 1]
```diff
@@ -25,6 +25,62 @@ provider "google" {
...
+  managed_zone = "${google_dns_managed_zone.test1_gpii_net.name}"
+  type         = "NS"
+  ttl          = 3600
+  rrdatas      = ["${google_dns_managed_zone.gcp_test1_gpii_net.name_servers}"]
+}
+
+resource "google_dns_managed_zone" "gcp_test1_gpii_net" {
+  name        = "gcp-test1-gpii-net"
+  dns_name    = "gcp.test1.gpii.net."
+  description = "gcp.test1.gpii.net DNS zone"
```
Comment (stepanstipl) [inline common/modules/gcp-dns/main.tf]: This zone does already exist.

**Comment by stepanstipl** [comment]
> Perhaps all the "${replace(var.organization_domain, "/^gpii.net/", "") == "" ? 1 : 0}" could be simplified to ${var.organization_domain == "gpii.net" ? 1 : 0}"?

@amatas did you see my comment above about simplifying the logic?

**Pull request merged**